sdk/python/flet/querystring.py

```diff
@@ -0,0 +1,104 @@
+import urllib.parse
+import re
+
+
+class UrlComponents:
+    """
+    `UrlComponents` are meant to be used internally for decoding-encoding, it has no external use
+    """
+
+    def _encode_url_component(self, url: str) -> str:
+        """
+        Function encodes querystring part of URL\n
+        Ex. q=dom & dogs -> q=dom+%26+dogs
+        """
+        return urllib.parse.quote(url)
+
+    def _decode_url_component(self, url: str) -> str:
+        """
+        Function decodes querystring part of URL\n
+        Ex. q=dom+%26+dogs -> q=dom & dogs
+        """
+        return urllib.parse.unquote(url)
+
+    def _is_encoded(self) -> bool:
+        """
+        Function returns True if URL is already encoded
+        """
+        if "?" in self.url:
+            q_result = self._querystring_part()
+            return (
+                True
+                if self._decode_url_component(
+                    self.url[q_result.start() + 1 : q_result.end()]
+                )
+                != self.url[q_result.start() + 1 : q_result.end()]
+                else False
+            )
+
+    def _querystring_part(self, url_string: bool = False):
+        """
+        Function sliced url part and returns querystring part.\n
+        Use case: checking querystring part for encode, assiging decoded value
+        """
+        pattern = re.compile(r"\?[\w\D]+")
+        data = pattern.search(self.url)
+        return data if url_string is False else self.url[data.start() + 1 : data.end()]
+

```

Comment:
Couldn't be query string extracted with just `urllib.parse.urlparse(self.url).query`, without using `re`, or I'm getting it wrong?